apps/researcher/src/lib/date-formatter/create-formatter.test.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  formatDate,
+  formatDateRange,
+  dateFormatSettings,
+} from './create-formatter';
+import {describe, expect, it} from '@jest/globals';
+
+const formatter = new Intl.DateTimeFormat('en', dateFormatSettings);
+
+const t = jest.fn(key => key);
+const formatDateTime = jest.fn(date => formatter.format(date));
+
+describe('formatDate', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('returns undefined if date is not provided', () => {
+    const result = formatDate({date: undefined, t, formatDateTime});
+    expect(result).toBeUndefined();
+    expect(t).not.toHaveBeenCalled();
+    expect(formatDateTime).not.toHaveBeenCalled();
+  });
+
+  it('formats the date with the date formatter for dates after 0', () => {
```

Comment:
The tests on line 25 and 31 cover different use cases:

1. Format with the date formatter for dates after 0
2. Format dates using translations for dates before 0

This does not include use cases such as:

1. Format with the date formatter for dates before 0
2. Format dates using translations for dates after 0

To be sure: is this correct?

Reply:
This is correct. Dates before 0 use the translations, and dates after 0 use the date format. So the two suggested tests will never happen. I can add a `expect(t).not.toHaveBeenCalled();` and `expect(formatDateTime).not.toHaveBeenCalled();` to the extisting tests to make that more clear.

Reply:
Yes, I think that would be a good thing, for clarity. But if you think that the tests look silly and don't add value, you could instead add a comment explaining why the tests do not cover certain (logical) use cases.